### ISSUE TITLE
Ignores differing ObjectProperty key/value whenever shorthand: true

### ISSUE THREAD

**Issue by jedwards1211** [benjamn/recast#704]
Obviously it's a programmer mistake to have `shorthand: true` whenever the key/value differ.  But let me make a case for why it's probably good to compare the key/value when printing anyway...

I have to do my codemod with @babel/traverse to guarantee that certain renamings respect `let`, `const`, `var`, etc scope.

In a test I was seeing `@babel/generate` output `{value: value0}` in one destructuring, but `recast.print` was outputting just `{value}`.  Looking at the AST, I saw that it was because `path.scope.rename` fails to clear the `shorthand` flag.

Seems like it would be worth comparing the `key` and `value` even on shorthand properties, just in case?

```
                      {
                        "type": "ObjectProperty",
                        "start": 56,
                        "end": 61,
                        "loc": {
                          "start": {
                            "line": 4,
                            "column": 20
                          },
                          "end": {
                            "line": 4,
                            "column": 25
                          }
                        },
                        "method": false,
                        "key": {
                          "type": "Identifier",
                          "start": 56,
                          "end": 61,
                          "loc": {
                            "start": {
                              "line": 4,
                              "column": 20
                            },
                            "end": {
                              "line": 4,
                              "column": 25
                            },
                            "identifierName": "value"
                          },
                          "name": "value"
                        },
                        "computed": false,
                        "shorthand": true,
                        "value": {
                          "type": "Identifier",
                          "start": 56,
                          "end": 61,
                          "loc": {
                            "start": {
                              "line": 4,
                              "column": 20
                            },
                            "end": {
                              "line": 4,
                              "column": 25
                            },
                            "identifierName": "value"
                          },
                          "name": "value0"
                        },
                        "extra": {
                          "shorthand": true
                        }
                      }
```

**Comment by eventualbuddha** [comment]
@jedwards1211 this would be fixed by #709. Look alright to you?

**Comment by eventualbuddha** [comment]
Fixed in v0.19.1.
